tests/arch/x86/arch_interface/src/main.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <sys/arch_interface.h>
+
+#define EFLAGS_IF_MASK 0X200
+/* Get the bit value of IRQ status */
+#define CHECK_IF_BIT(x, y) {        \
+	x = irq_lock();                 \
+	y = (x & EFLAGS_IF_MASK);       \
+	irq_unlock(x);                  \
+}
+static struct k_timer idle_timer;
+static void idle_timer_expiry_function(struct k_timer *timer_id)
+{
+	/* nothing to do */
+}
+
+/**
+ * @brief Verify arch interface of timing functions
+ *
+ * @details Initializing arch timing and start it,
+ * then catch the freq value, it is successful if
+ * the return value is non-zero, and verify
+ * arch_timing_cycles_to_ns_avg API. Then stop it.
+ *
+ * @see arch_timing_freq_get_mhz()
+ * arch_timing_cycles_to_ns_avg()
+ */
+void test_arch_timing_func(void)
+{
+	uint32_t freq = 0;
+	uint64_t cycle_to_ns = 0;
+
+	arch_timing_init();
+	arch_timing_start();
```

Comment:
Why the need to test the arch interface for timing functions, and not the timing functions themselves? The timing functions will call into the arch timing functions, so there is no need for separate tests for these.

Reply:
Hi @dcpleung, I re-designed the test case for timing, could you help to review this? Thanks.

Reply:
Usage of timing functions should have been covered by `tests/benchmarks/latency_measure`. Please consolidate this into that test.

Reply:
OK, thanks.